### DISCUSSION TITLE
Validation champ date, calcul 1 an max

### DISCUSSION THREAD

**Discussion by jodionrq** [MTESSDev/FRW#123] [Annonces]
Bonjour,
Je dois ajouter une validation dans un champ date pour que la date inscrite ne dépasse pas un an par rapport à la date d'aujourd'hui.
Exemple, si on inscrit 08/31/2024, je devrais avoir une erreur.
Est-ce qu'on peut utiliser une validation simple de type "before" (sans mettre de date précise pour calculer un an, comme un getFullYear)?
Ou alors, est-ce qu'on doit utiliser une validation de type custom? Si oui, l'exemple dans le P700U s'applique à un champs texte au format YYYY. J'ai tenté d'utiliser cet exemple avec mon champ de type date mais sans succès.
Je peux vous fournir mon exemple au besoin.
Merci!

**Comment by jodionrq** [comment]
Pour l'instant, nous avons réussi à faire fonctionner le calcul, mais le message de validation n'est pas pris en compte comme nous voulons. Tout ce qui apparait c'est "Ce champ n'est pas valide."
Voici le code : 

                    - type: date
                      name: date90jours
                      label: 
                        fr: Date à laquelle vous avez mis ou prévoyez mettre fin à votre emploi.
                      validations:                        
                        maxUnAnCustom: 1
                      additionals:    
                        :validation-rules: |
                          {
                            maxUnAnCustom: ({ value }) => {
                              const currentDate = new Date();
                              const selectedDate = new Date(value);

                              // Ajouter un an à la date actuelle
                              currentDate.setFullYear(currentDate.getFullYear() + 1);

                              return selectedDate <= currentDate;
                            }
                          } 
                        #Problème avec de message avec la validation, le message actuel qui apparait est 'ce champ n'est pas valide'.
                        validation-messages:
                          maxUnAnCustom:
                            fr: |
                              ({ name }) => {
                                return `La valeur du champ « ${name} » doit être inférieure d'un an à la date d'aujourd'hui.`;
                              }

Savez-vous ce qui fait que le message ne s'affiche pas comme il faut?

**Comment by j0ber** [comment]
Bonjour, 
À l'oeil, "validation-messages" semble au même niveau que ":validation-rules"... YAML = indentation IMPORTANTE :)  

Normalement "validation-messages" devrait être au même niveau que additionnals. Regarder l'exemple https://formulaires.it.mtess.gouv.qc.ca/fr/Form/7/P700U/0/N/#p=validationsCustom

**Comment by jodionrq** [comment]
Bonjour,
Excellent, c'était une simple question d'alignement.
Merci beaucoup!
